### PR TITLE
Issue #42 feat: New Match Type Question Implementation

### PR DESCRIPTION
**Pull request by gupta-arpan**
UI created:

Screenshot from 2023-08-10 15-33-12

https://github.com/Sunbird-inQuiry/editor/assets/98040726/41f6080a-8e3b-4c03-80c6-e1e3a071dbf6

### CODE REVIEW

File: projects/questionset-editor-library/src/lib/components/match/match.component.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+@Component({
+  selector: 'lib-match',
+  templateUrl: './match.component.html',
+  styleUrls: ['./match.component.scss']
+})
+export class MatchComponent implements OnInit, OnChanges {
+  @Input() editorState: any;
+  @Input() showFormError;
+  @Input() sourcingSettings;
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/match/match.component.ts:14]: @gupta-arpan `sourcingSettings` is not getting used anywhere. Please remove it from this component.

File: projects/questionset-editor-library/src/lib/components/match/match.component.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  @Input() questionPrimaryCategory;
+  @Input() mapping = [];
+  @Input() isReadOnlyMode;
+  @Input() maxScore;
+  @Output() editorDataOutput: EventEmitter<any> = new EventEmitter<any>();
+  public setCharacterLimit = 160;
+  public setImageLimit = 1;
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/match/match.component.ts:21]: `setImageLimit` is not getting used anywhere.

File: projects/questionset-editor-library/src/lib/components/match/match.component.html
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/match/match.component.html:1]: Please format each tags, It's not easy to read.

File: projects/questionset-editor-library/src/lib/components/match/match.component.spec.data.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        left: "<p>Mango</p>",
+        right: "<p>Fruit</p>",
+      },
+    ],
+    templateId: "mtf-vertical",
+    correctMatchPair: [{ "0": 0 }, { "1": 1 }],
+    numberOfOptions: 4,
+  },
+  prepareMtfBody: {
+    templateId: "default",
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/match/match.component.spec.data.ts:19]: change the mock value of `templateId` to either 'mtf-horizontal' or 'mtf-vertical'

File: projects/questionset-editor-library/src/lib/components/question/question.component.html [hunk 4]
```diff
@@ -64,6 +64,13 @@
                 (editorDataOutput)="editorDataHandler($event)"
                 [sourcingSettings]="sourcingSettings" [mapping]="scoreMapping" [maxScore]="maxScore"
                 [isReadOnlyMode]="isReadOnlyMode"></lib-options>
+              
+              <lib-match *ngIf="questionInteractionType === 'match'"
+                [questionPrimaryCategory]="questionPrimaryCategory"
+                [editorState]="editorState" [showFormError]="showFormError"
+                (editorDataOutput)="editorDataHandler($event)"
+                [sourcingSettings]="sourcingSettings" [mapping]="scoreMapping" [maxScore]="maxScore"
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/question/question.component.html:72]: remove `[sourcingSettings]="sourcingSettings"`

File: projects/questionset-editor-library/src/lib/components/question/question.component.ts [hunk 5]
```diff
@@ -243,6 +244,31 @@ export class QuestionComponent implements OnInit, AfterViewInit, OnDestroy {
                   this.editorState.responseDeclaration = _.get(this.questionMetaData, 'responseDeclaration');
                 }
               }
+
+              if (this.questionInteractionType === 'match') {
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/question/question.component.ts:248]: Lots of lines inside this if condition is similar to 'choice' condition. Due to this code duplication is happening. Please generalise this code.

Comment (gupta-arpan) [reply projects/questionset-editor-library/src/lib/components/question/question.component.ts]: Completed

File: projects/questionset-editor-library/src/lib/components/match/match.component.spec.data.ts (hunk 3 above)
Comment (gupta-arpan) [reply projects/questionset-editor-library/src/lib/components/match/match.component.spec.data.ts]: completed

File: projects/questionset-editor-library/src/lib/components/match/match.component.html
Comment (gupta-arpan) [reply projects/questionset-editor-library/src/lib/components/match/match.component.html:1]: completed

File: projects/questionset-editor-library/src/lib/components/match/match.component.ts (hunk 2 above)
Comment (gupta-arpan) [reply projects/questionset-editor-library/src/lib/components/match/match.component.ts]: completed

File: projects/questionset-editor-library/src/lib/components/match/match.component.ts (hunk 1 above)
Comment (gupta-arpan) [reply projects/questionset-editor-library/src/lib/components/match/match.component.ts]: completed

**Review by rajnishdargan — APPROVED** [review]

**Pull request merged**